File: server/app/models/workspace.py
"""工作区数据模型"""
from datetime import datetime
from typing import Optional
from pydantic import BaseModel, Field


class WorkspaceCreate(BaseModel):
    """创建工作区请求"""
    name: str = Field(..., min_length=1, max_length=100, description="工作区名称")
    description: Optional[str] = Field(None, max_length=500, description="工作区描述")


class WorkspaceInfo(BaseModel):
    """工作区信息响应"""
    id: str
    name: str
    description: Optional[str] = None
    created_at: datetime
    updated_at: datetime
    path: str
    deleted: bool = False
    deleted_at: Optional[datetime] = None


class Workspace(BaseModel):
    """工作区完整模型"""
    id: str
    name: str
    description: Optional[str] = None
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True

File: mobile/lib/models/workspace.dart
/// 工作区模型（完整版，用于详细视图）
class Workspace {
  final String id;
  final String name;
  final String? description;
  final DateTime createdAt;
  final DateTime updatedAt;
  final String path;
  final bool deleted;
  final DateTime? deletedAt;

  Workspace({
    required this.id,
    required this.name,
    this.description,
    required this.createdAt,
    required this.updatedAt,
    required this.path,
    this.deleted = false,
    this.deletedAt,
  });

  factory Workspace.fromJson(Map<String, dynamic> json) {
    return Workspace(
      id: json['id'] as String,
      name: json['name'] as String,
      description: json['description'] as String?,
      createdAt: DateTime.parse(json['created_at'] as String),
      updatedAt: DateTime.parse(json['updated_at'] as String),
      path: json['path'] as String,
      deleted: json['deleted'] as bool? ?? false,
      deletedAt: json['deleted_at'] != null
          ? DateTime.parse(json['deleted_at'] as String)
          : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'description': description,
      'created_at': createdAt.toIso8601String(),
      'updated_at': updatedAt.toIso8601String(),
      'path': path,
      'deleted': deleted,
      'deleted_at': deletedAt?.toIso8601String(),
    };
  }
}

/// 工作区信息（简化版，用于列表展示）
class WorkspaceInfo {
  final String id;
  final String name;
  final String? description;
  final String path;
  final bool deleted;

  WorkspaceInfo({
    required this.id,
    required this.name,
    this.description,
    required this.path,
    this.deleted = false,
  });

  factory WorkspaceInfo.fromJson(Map<String, dynamic> json) {
    return WorkspaceInfo(
      id: json['id'] as String,
      name: json['name'] as String,
      description: json['description'] as String?,
      path: json['path'] as String,
      deleted: json['deleted'] as bool? ?? false,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'description': description,
      'path': path,
      'deleted': deleted,
    };
  }
}
